### ISSUE TITLE
abstract over rand::Rng

### ISSUE THREAD

**Issue by neadeer** [Kromey/fast_poisson#13]
it would be nice if we could let the caller pass in their own preferred (or required) R: rand::Rng

this would  also allow the caller to initialize their rng with a seed once and use the same rng for other parts of a deterministic initialization. forcing us to use a separate seed for each poisson disk doesn't compose as well

this would also allow users who don't use our blessed rng to avoid that dependencies, as we wouldn't have to bake in our rng as a dependency

**Comment by Kromey** [comment]
When I originally needed a Poisson disk distribution, I found the [`poisson`](https://docs.rs/poisson/latest/poisson/) crate and tried to use that. The problem is that that crate abstracts over `rand::Rng`, however in Rust traits are *not the same* across versions: Since I was using `rand` 0.8.3, but that crate was using version 0.6.5, I could not pass in any generator at all. This rendered the crate wholly useless, as I needed the newer version of `rand` for functionality I was using elsewhere in my project. I initially tried forking that one and just updating the `rand` version, but so much has changed that it would have been nearly a complete rewrite.

That's why when I created this one I deliberately chose to forego the option to supply your own RNG, so as to provide a crate that will be functional for as many people as possible, using whatever version of `rand` - or even an entirely different crate altogether, or nothing at all - they want, even if I neglect to keep up to date with the latest versions of that crate.
